mlflow/store/_unity_catalog/registry/rest_store.py

```diff
@@ -648,6 +660,7 @@ def create_model_version(
                     tags=uc_model_version_tag_from_mlflow_tags(tags),
                     run_tracking_server_id=source_workspace_id,
                     feature_deps=feature_deps,
+                    vector_index_deps = vector_index_deps
```

Comment:
You'll need to add this to the proto definition in `mlflow/protos/databricks_uc_registry_messages.proto` and rerun `dev/generate-protos.sh`.